lib/bolt/config.rb

```diff
@@ -189,9 +305,13 @@ def initialize(project, config_data, overrides = {})
         'transport'           => 'ssh'
       }
 
-      loaded_data = config_data.map do |config|
-        @config_files.push(config[:filepath])
-        config[:data]
+      loaded_data = config_data.each_with_object([]) do |data, acc|
+        @warnings = @warnings.concat(data[:warnings]) if data[:warnings]
```

Comment:
`concat` already modifies the object directly, so this shouldn't need to reassign `@warnings`

Reply:
And since several files we load will always have this be empty for now (like project-level bolt.yaml) I think we can either not include `warnings` in those data hashes and keep the check here for nil, or make this `data[:warnings].empty?`.